packages/core/custom-branding/core-custom-branding-server-internal/custom_branding_service.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type {
+  CustomBrandingStart,
+  CustomBrandingStartDeps,
+} from '@kbn/core-custom-branding-server';
+import type { CustomBranding } from '@kbn/core-custom-branding-common';
+import { KibanaRequest } from '@kbn/core-http-server';
+import { InternalSavedObjectsServiceStart } from '@kbn/core-saved-objects-server-internal';
+import { InternalUiSettingsServiceStart } from '@kbn/core-ui-settings-server-internal';
+import type { CoreContext } from '@kbn/core-base-server-internal';
+import type { Logger } from '@kbn/logging';
+import { IUiSettingsClient } from '@kbn/core-ui-settings-server';
+
+/**
+ * @internal
+ */
+export interface InternalCustomBrandingSetup {
+  register: (pluginName: string) => void;
+}
+
+export class CustomBrandingService {
+  private pluginName: string | undefined;
+  private savedObjects?: InternalSavedObjectsServiceStart;
+  private uiSettings?: InternalUiSettingsServiceStart;
+  private readonly settingsKeys: Array<keyof CustomBranding> = [
+    'logo',
+    'customizedLogo',
+    'faviconPNG',
+    'faviconSVG',
+    'pageTitle',
+  ];
+  private logger: Logger;
+
+  constructor(coreContext: CoreContext) {
+    this.logger = coreContext.logger.get('custom-branding-service');
+  }
+
+  private getBrandingFrom = async (uiSettingsClient: IUiSettingsClient) => {
+    const branding: CustomBranding = {};
+    for (let i = 0; i < this.settingsKeys!.length; i++) {
+      const key = this.settingsKeys[i];
+      const fullKey = `customBranding:${key}`;
+      const value = await uiSettingsClient.get(fullKey);
+      if (value) {
+        branding[key] = value;
+      }
+    }
+    return branding;
+  };
```

Comment:
This logic should live in the `customBranding` plugin, as the way the custom branding values are retrieved should remain an implementation detail of the plugin.

The `customBranding` plugin should register this logic via the `register` setup function.

```ts
CustomBrandingPlugin {

    setup(core: CoreSetup) {
       core.customBranding.register( async (request) => {
            const [coreStart] = await core.getStartServices();
            const soService = coreStart.savedObjects.getScopedClient(request);
            // .....
            return customBranding.
       })
     }
}


```